tee-worker/app-libs/parentchain-interface/src/integritee/mod.rs

```diff
@@ -0,0 +1,227 @@
+/*
+	Copyright 2021 Integritee AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+mod event_filter;
+mod event_handler;
+mod extrinsic_parser;
+
+use crate::{
+	decode_and_log_error,
+	indirect_calls::{
+		ActivateIdentityArgs, DeactivateIdentityArgs, InvokeArgs, LinkIdentityArgs,
+		RemoveScheduledEnclaveArgs, RequestVCArgs, ShieldFundsArgs, UpdateScheduledEnclaveArgs,
+	},
+	integritee::extrinsic_parser::ParseExtrinsic,
+};
+use codec::{Decode, Encode};
+use core::marker::PhantomData;
+pub use event_filter::FilterableEvents;
+pub use event_handler::ParentchainEventHandler;
+pub use extrinsic_parser::ParentchainExtrinsicParser;
+use ita_stf::TrustedCallSigned;
+use itc_parentchain_indirect_calls_executor::{
+	error::{Error, Result},
+	filter_metadata::FilterIntoDataFrom,
+	IndirectDispatch,
+};
+use itp_node_api::metadata::NodeMetadataTrait;
+use itp_stf_primitives::traits::IndirectExecutor;
+use itp_types::{CallIndex, H256};
+use log::trace;
+use sp_core::crypto::AccountId32;
+use sp_runtime::MultiAddress;
+use sp_std::vec::Vec;
+
+/// The default indirect call (extrinsic-triggered) of the Integritee-Parachain.
+#[derive(Debug, Clone, Encode, Decode, Eq, PartialEq)]
+pub enum IndirectCall {
+	#[codec(index = 0)]
+	ShieldFunds(ShieldFundsArgs),
+	#[codec(index = 1)]
+	Invoke(InvokeArgs),
+	// Litentry
+	#[codec(index = 2)]
+	LinkIdentity(LinkIdentityArgs, Option<MultiAddress<AccountId32, ()>>, H256),
+	#[codec(index = 3)]
+	DeactivateIdentity(DeactivateIdentityArgs, Option<MultiAddress<AccountId32, ()>>, H256),
+	#[codec(index = 4)]
+	ActivateIdentity(ActivateIdentityArgs, Option<MultiAddress<AccountId32, ()>>, H256),
+	#[codec(index = 5)]
+	RequestVC(RequestVCArgs, Option<MultiAddress<AccountId32, ()>>, H256),
```

Comment:
`tee-worker/app-libs/parentchain-interface/src/integritee/mod.rs`

What do you think about renaming this file or creating separate one for litentry parentchain and removing all litentry related stuff from this?

Reply:
Thanks, I'll create a `litentry` mod next to `integritee` (and revert all changes in the current `integritee` mod)

Reply:
During the refactoring process I realised there's more than that - including many enclave initialisation process.

I'll create a separate issue for it, and maybe have a discussion beforehand if we should rename all